src/config.c

```diff
@@ -164,74 +257,290 @@ static void _Config_SetToDefaults(RedisModuleCtx *ctx) {
+			// result-set size
+			//----------------------------------------------------------------------
+
+		case Config_RESULTSET_MAX_SIZE:
+			{
+				long long resultset_max_size;
+				if(!_Config_ParseInteger(val, &resultset_max_size)) return false;
+
+				Config_resultset_max_size_set(resultset_max_size);
+			}
+			break;
+
+			//----------------------------------------------------------------------
+			// maintain transpose
+			//----------------------------------------------------------------------
+
+		case Config_MAINTAIN_TRANSPOSE:
+			{
+				bool maintain_transpose;
+				if(!_Config_ParseYesNo(val, &maintain_transpose)) return false;
+
+				Config_maintain_transpose_set(maintain_transpose);
+			}
+			break;
+
+			//----------------------------------------------------------------------
+			// virtual key entity count
+			//----------------------------------------------------------------------
+
+		case Config_VKEY_MAX_ENTITY_COUNT:
+			{
+				long long vkey_max_entity_count;
+				if(!_Config_ParsePositiveInteger(val, &vkey_max_entity_count)) return false;
+
+				Config_virtual_key_entity_count_set(vkey_max_entity_count);
+			}
+			break;
+
+			//----------------------------------------------------------------------
+			// async deleteion
+			//----------------------------------------------------------------------
+
+		case Config_ASYNC_DELETE:
+			{
+				bool async_delete;
+				if(!_Config_ParseYesNo(val, &async_delete)) return false;
+
+				Config_async_delete_set(async_delete);
+			}
+			break;
+
+        //----------------------------------------------------------------------
+        // invalid option
+        //----------------------------------------------------------------------
+
+        default : 
+			ASSERT("invalid option field" && false);
+            return false;
+    }
+
+	return true;
 }
 
-uint64_t Config_GetCacheSize() {
-	return config.cache_size;
+bool Config_Option_get(Config_Option_Field field, ...) {
+
+	//--------------------------------------------------------------------------
+	// get the option
+	//--------------------------------------------------------------------------
+
+	va_list ap;
+
+	switch (field)
+	{
+		//----------------------------------------------------------------------
+		// cache size
+		//----------------------------------------------------------------------
+
+		case Config_CACHE_SIZE:
+			{
+				va_start(ap, field);
+				uint64_t *cache_size = va_arg(ap, uint64_t*);
+				va_end(ap);
+
+				if(cache_size == NULL) return false;
+				(*cache_size) = Config_cache_size_get();
+			}
+			break;
+
+			//----------------------------------------------------------------------
+			// OpenMP thread count
+			//----------------------------------------------------------------------
+
+		case Config_OPENMP_NTHREAD:
+			{
+				va_start(ap, field);
+				uint *omp_nthreads = va_arg(ap, uint*);
+				va_end(ap);
+
+				if(omp_nthreads == NULL) return false;
```

Comment:
These guards seem overly permissive to me. If any of these values are returned as NULL, that's an error and should be asserted on.